client/app/scripts/superdesk/itemList/itemList.js

```diff
@@ -0,0 +1,287 @@
+                    default_operator: 'AND'
+                }
+            };
+        }
+        // process search
+        if (options.search) {
+            var queryContentAny = [];
+            _.each(_.values(fields), function(dbField) {
+                queryContentAny.push(dbField + ':(' + options.search + ')');
+            });
+            query.source.query.filtered.query = {
+                query_string: {
+                    query: queryContentAny.join(' '),
+                    lenient: false,
+                    default_operator: 'OR'
+                }
+            };
+        }
+
+        return query;
+    }
+
+    this.fetch = function(options) {
+        options = _.extend({}, DEFAULT_OPTIONS, options);
+        var query = getQuery(options);
+        return api(options.endpoint, options.endpointParam || undefined)
+            .query(query)
+            .then(function(result) {
+                return result;
+            });
+    };
+}])
+.provider('ItemList', function() {
+    this.$get = ['itemListService', function(itemListService) {
+        var ItemList = function() {
+            this.listeners = [];
+            this.options = _.clone(DEFAULT_OPTIONS);
+            this.result = {};
+            this.maxPage = 0;
+        };
+
+        ItemList.prototype.setOptions = function(options) {
+            this.options = _.extend(this.options, options);
+            return this;
+        };
+
+        ItemList.prototype.addListener = function(listener) {
+            this.listeners.push(listener);
+            return this;
+        };
+
+        ItemList.prototype.removeListener = function(listener) {
+            _.remove(this.listeners, function(i) {
+                return i === listener;
+            });
+            return this;
+        };
+
+        ItemList.prototype.fetch = function() {
+            var self = this;
+
+            return itemListService.fetch(this.options)
+            .then(function(result) {
+                self.result = result;
+                self.maxPage = Math.ceil(result._meta.total / self.options.pageSize) || 0;
+                _.each(self.listeners, function(listener) {
+                    listener(result);
+                });
+                return result;
+            });
+        };
+
+        return ItemList;
+    }];
+})
+.directive('sdItemListWidget', ['ItemList', 'preferencesService', 'notify', function(ItemList, preferencesService, notify) {
+    return {
+        scope: {
+            options: '=',
+            itemListOptions: '=',
+            actions: '='
+        },
+        templateUrl: 'scripts/superdesk/itemList/views/item-list-widget.html',
+        link: function(scope, element, attrs) {
+            scope.items = null;
+            scope.maxPage = 1;
+            scope.pinnedItems = [];
+            scope.selected = null;
+
+            var pinnedList = {};
+            var itemList = new ItemList();
+
+            itemList.addListener(function() {
```

Comment:
there is removeListener, and it can be used to unregister listeners. But in most cases it won't be needed, as it will die with the scope anyway.